backend/pkg/api/handle_schema_registry.go

```diff
@@ -817,3 +819,33 @@ func (api *API) handleValidateSchema() http.HandlerFunc {
 		rest.SendResponse(w, r, api.Logger, http.StatusOK, res)
 	}
 }
+
+func getSubjectFromRequestPath(r *http.Request) string {
+	// subject extraction gets complicated
+	// With a path like /api/schema-registry/subjects/%252F/versions/latest
+	// r.URL.Path is /api/schemas/schema-registry/%2F/versions/latest
+	// while RequestURI is /api/schemas/schema-registry/%252F/versions/latest
+	// that means that chi.URLParam() returns  %2F
+	// which then url.PathUnescape() within rest.GetURLParam() returns "/"
+	// which is "double escaped"
+	// so we have to manually extract and escape the subject part ourselves
+
+	requestURI := r.RequestURI
+	// remove the api route prefix
+	requestURI = strings.ReplaceAll(requestURI, "/api/schema-registry/subjects/", "")
+	requestURI = strings.ReplaceAll(requestURI, "/api/schema-registry/config/", "")
+	// find the versions suffix of the path
+	subjectPart := requestURI
+	lastIndex := strings.Index(requestURI, "/")
+	if lastIndex > 0 { // satisfy linter
+		// and extract the subject at the start of the string
+		subjectPart = requestURI[:lastIndex]
+	}
+
+	subject, err := url.PathUnescape(subjectPart)
+	if err != nil {
+		subject = subjectPart
+	}
+
+	return subject
+}
```

Comment:
@bojand I actually do not understand this, can you add a table unit test with expect inputs & outputs? That would help me understand this better in the future I believe